.github/workflows/tag-maintainers.yml

```diff
@@ -0,0 +1,235 @@
+          echo "EOF" >> $GITHUB_OUTPUT
+
+          echo "users_to_exclude<<EOF" >> $GITHUB_OUTPUT
+          echo "$USERS_TO_EXCLUDE" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+          echo "Current pending reviewers: $PENDING_REVIEWERS"
+          echo "Complete list of users to exclude: $USERS_TO_EXCLUDE"
+
+      # Step 8: Filter the maintainer list to only include repository collaborators.
+      # You can only request reviews from users with write access/collaborators.
+      - name: Check maintainer collaborator status
+        id: check-collaborators
+        env:
+          GH_TOKEN: ${{ steps.app-token.outputs.token || secrets.GITHUB_TOKEN }}
+          MAINTAINERS: ${{ steps.extract-maintainers.outputs.maintainers }}
+          USERS_TO_EXCLUDE: ${{ steps.current-reviewers.outputs.users_to_exclude }}
+          REPO: "${{ github.repository }}"
+        run: |
+          NEW_REVIEWERS=()
+
+          # If there are no maintainers, exit early.
+          if [[ -z "$MAINTAINERS" ]]; then
+            echo "No maintainers to check."
+            echo "new_reviewers=" >> "$GITHUB_OUTPUT"
+            exit 0
+          fi
+
+          for MAINTAINER in $MAINTAINERS; do
+            # Skip if the user is already in the exclusion list (pending/past reviewer).
+            if echo "$USERS_TO_EXCLUDE" | grep -q -w "$MAINTAINER"; then
+              echo "$MAINTAINER is already involved in the review, skipping."
+              continue
+            fi
+
+            echo "Checking if $MAINTAINER is a collaborator..."
+            if gh api "/repos/$REPO/collaborators/$MAINTAINER" --silent; then
+              echo "User $MAINTAINER is a collaborator, adding to new reviewers list."
+              NEW_REVIEWERS+=("$MAINTAINER")
+            else
+              echo "User $MAINTAINER is not a repository collaborator, skipping."
+            fi
+          done
+
+          NEW_REVIEWERS_LIST=$(printf "%s " "${NEW_REVIEWERS[@]}")
+          echo "new_reviewers=${NEW_REVIEWERS_LIST% }" >> "$GITHUB_OUTPUT"
+          echo "New reviewers to add: ${NEW_REVIEWERS_LIST% }"
+
+      # Step 9: Remove reviewers who are no longer maintainers of the changed files.
+      - name: Remove outdated reviewers
+        env:
+          GH_TOKEN: ${{ steps.app-token.outputs.token || secrets.GITHUB_TOKEN }}
+          MAINTAINERS: ${{ steps.extract-maintainers.outputs.maintainers }}
+          PENDING_REVIEWERS: ${{ steps.current-reviewers.outputs.pending_reviewers }}
+        run: |
+          remove_reviewers() {
+            local reviewers_to_remove="$1"
+            local reason="$2"
+            for REVIEWER in $reviewers_to_remove; do
+              echo "Removing review request from $REVIEWER ($reason)"
+              gh api --method DELETE "/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/requested_reviewers" \
+                --input - <<< "{\"reviewers\": [\"$REVIEWER\"]}"
```

Comment:
My gut doesn't like the idea of constructing JSON with a bash string... I'm guessing GitHub usernames can't contain things like `"` so it's probably fine. But I'd be more comfortable using `jq` to marshall the JSON.

```
jq -c '{ reviewers: [.] }'
```